pandas/tests/io/excel/test_writers.py

```diff
@@ -1115,6 +1115,29 @@ def test_bytes_io(self, engine):
             reread_df = pd.read_excel(bio, index_col=0)
             tm.assert_frame_equal(df, reread_df)
 
+    def test_engine_kwargs(self, engine, path):
+        df = DataFrame([{"A": 1, "B": 2}, {"A": 3, "B": 4}])
+
+        msgs = {
+            "odf": r"OpenDocumentSpreadsheet() got an unexpected keyword "
+            r"argument 'foo'",
+            "openpyxl": r"load_workbook() got an unexpected keyword argument 'foo'",
+            "xlsxwriter": r"__init__() got an unexpected keyword argument 'foo'",
+        }
+
+        # Handle change in error message for openpyxl (write and append mode)
+        if engine == "openpyxl" and os.path.exists(path):
+            msgs[
```

Comment:
Shouldn't this path test something?

Reply:
@mroeschke I added that IF statement to check for an existing file so that I can test the openpyxl engine's append mode.

If the file exists, I know that I need to update the error message so that it aligns with the error message that I would receive if I were to pass invalid arguments to `openpyxl.load_workbook()`


![image](https://github.com/pandas-dev/pandas/assets/45905457/9a5a8058-aae1-4f2f-936d-fc04d01346f9)

Reply:
This if statement just updates the dict and doesn't check anything else though

Reply:
That's correct. Is that a problem? The checking of `msgs[engine]` is done below the if statement under the else statement. 

Would it make more sense if I bump everything within the **else** statement to the left and then remove **else**? The IF statement is necessary so that I can dynamically update `msgs["openpyxl"]` when I need to check the append mode.

Reply:
Sorry about that. The issue should be fixed in my most recent push!